runtime/shared_common/CacheMap.cpp

```diff
@@ -6327,9 +6327,6 @@ SH_CacheMap::matchAotMethod(MethodSpecTable* specTable, IDATA numSpecs, J9UTF8*
 		if (NULL == specTable[i].methodSig) {
 			/* Any signature */
 			sigMatch = true;
-		} else if (NULL == J9UTF8_DATA(romMethodSig)) {
```

Comment:
@hangshao0 do you think the test should be fixed to do what was intended rather than just removed? i.e. it could check if the length is zero.

Reply:
If the length is zero it seems the following code will search random memory for '(' and ')' and then use what it finds to try and make a match.

Reply:
I am not worried about this special case much, but changing it to test on the UTF8 length looks safer to me.

Reply:
I think a bigger concern is that the data in `J9UTF8` need not be null-terminated, and so using `strchr()` is not reasonable.

Reply:
Ya, it should be updated to use strnstr().

Reply:
Or sth like `memchr()`.

Reply:
If we use either `strnstr()` or `memchr()`, we need to also pass in a length over which to search. What value should I be passing in for the length?

Reply:
There is a`J9UTF8_LENGTH` macro